src/main/java/hanshyn/onlinebookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,97 @@
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCartDto getShoppingCart(Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId()).orElseThrow(
+                () -> new RuntimeException("Can't found user: user")
+        );
+
+        Set<CartItem> cartItem = cartItemRepository.findByShoppingCartId(shoppingCart.getId());
+
+        Set<CartItemsResponseDto> cartItemsResponseDtos = cartItem.stream()
+                .map(cartItemMapper::toDo)
+                .collect(Collectors.toSet());
+
+        ShoppingCartDto shoppingCartDto = shoppingCartMapper.toDto(shoppingCart);
+        shoppingCartDto.setCartItems(cartItemsResponseDtos);
+
+        return shoppingCartDto;
+    }
+
+    @Override
+    public CartItemsResponseDto saveBookToShoppingCart(Authentication authentication,
+                                                       CartItemsRequestDto cartItemsRequestDto) {
+        User user = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId())
+                .orElseGet(() -> createNewShoppingCart(user));
+
+        Book book = bookRepository.findById(cartItemsRequestDto.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException("Book by id:"
+                        + cartItemsRequestDto.getBookId() + "not found"));
+
+        CartItem cartItem = cartItemMapper.toModel(cartItemsRequestDto);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItem.setBook(book);
+
+        CartItem saveCartItem = cartItemRepository.save(cartItem);
+        Set<CartItem> cartItems = cartItemRepository.findByShoppingCartId(shoppingCart.getId());
+        cartItems.add(saveCartItem);
+        shoppingCart.setCartItems(cartItems);
+        shoppingCartRepository.save(shoppingCart);
+        return cartItemMapper.toDo(cartItem);
+    }
+
+    private ShoppingCart createNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        return shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public CartItemsResponseDto updateQuantity(CartItemsUpdateRequestDto cartItemsUpdateRequestDto,
+                                               Long id) {
+
+        CartItem cartItem = cartItemRepository.findById(id).orElseThrow(
+                () -> new RuntimeException("Can't found cart item:by id " + id));
+
+        cartItem.setQuantity(cartItemsUpdateRequestDto.getQuantity());
+
+        return cartItemMapper.toDo(cartItemRepository.save(cartItem));
+    }
+
+    @Override
+    public void deleteCartItems(Long id) {
+        cartItemRepository.deleteById(id);
+    }
```

Comment:
On add operation you check user that is authenticated but on update/delete you allow anyone to modify other user's cart. 
E.g. I'm user 1 and have items 1, 2 and 3 in my cart. I'm sending request to delete cart-item with id 4. Then you will delete it for some other user obviously